### PR TITLE
[Manta-PC] Integrate runtime upgrade test into CI

### PR DESCRIPTION
**Pull request by ghzlatarev**
## Description

closes https://github.com/Manta-Network/pallet-manta-pay/issues/92

* Integrates a runtime upgrade test into the CI
* First the spec version from the current commit is compared to the spec verion in the base branch.
* If they don't match the runtime upgrade job is triggered.
* The code from the base branch is used to launch the tesntet
* The code from the current commit is used to build the wasm to upgrade the runtime
* The log for the runtime upgrade is parsed for success and also block production is checked.
* The log is also uploaded as artifact.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (`manta` or `manta-pc`) with right title (start with [Manta] or [Manta-PC])
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] If needed, notify the committer this is a draft-release and a tag is needed after merging the PR.

### CODE REVIEW

File: runtime/calamari/src/lib.rs
Comment (ghzlatarev) [inline runtime/calamari/src/lib.rs:85]: TODO: only for testing, revert when done.

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml]: These secrets here are only used for the CI.

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml:1318]: This is a workaround for an AWS error MaxSpotInstanceCountExceeded: Max spot instance count exceeded

TODO: revert before merging.

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml:357]: This has to be a hash with old version of the code.
Maybe we can hard-code it as manta-pc or let it be an input for the developer.

This current hash is a commit which has the correct accounts for the chain spec, as ``manta-pc`` doesn't have them yet.

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml]: Not sure if we should leave this as `manta-pc` or let it be an input for the developer.

The commented hash is a commit which has the correct accounts for the chain spec, as ``manta-pc`` doesn't have them yet. Used it to trigger the last CI build.

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml]: These conditionals need to be on each step of every job, unfortunately.
There's no other way to exit early with success.
We need jobs to rely on the runtime upgrade stuff only if the runtime upgrade actually happens.

**Comment by ghzlatarev** [comment]
The last successful run before I reverted the chain spec bump, which was triggering it ---> https://github.com/Manta-Network/Manta/actions/runs/1195024749

**Comment by stechu** [comment]
Can you fix the PR name?

File: .github/workflows/publish-draft-releases.yml
Comment (stechu) [inline .github/workflows/publish-draft-releases.yml]: can you use a .json file instead of hardcode this in the yml? maybe create `resources/local-deployment.json`.

File: .github/workflows/publish-draft-releases.yml
Comment (stechu) [reply .github/workflows/publish-draft-releases.yml]: Can you elaborate about the secrets usage here?

**Review by stechu — CHANGES_REQUESTED** [review]
LGTM in general. Please see the detailed comments. Also, I would wait for Rob's review for this PR before merging.

**Comment by stechu** [comment]
I also noticed that there are some hardcoded values, e.g:

https://github.com/Manta-Network/Manta/pull/180/files#diff-84486036e95f7e22168734603cbdd311736c85fb23a0d0e0660988a67d3605ecR1315

I think we should make it not hardcoded (maybe with the exception of AWS instance types).

**Review by grenade — APPROVED** [review]
lgtm

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [reply .github/workflows/publish-draft-releases.yml]: We don't have a good way to handle these secrets in the CI. We need an account with a gpg key to retrieve them from the pass store. 
So for now I just generated a new set of accounts/secrets to only be used for CI

**Comment by ghzlatarev** [comment]
> Can you fix the PR name?

Done

File: .github/workflows/publish-draft-releases.yml
Comment (stechu) [reply .github/workflows/publish-draft-releases.yml]: are these testnet secrets? or some other secrets?

Comment (ghzlatarev) [reply .github/workflows/publish-draft-releases.yml]: Completely new that I generated only for this job and ``genesis/calamari-testnet-ci-genesis.json`

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [reply .github/workflows/publish-draft-releases.yml]: Yea, can do.

**Comment by ghzlatarev** [comment]
> I also noticed that there are some hardcoded values, e.g:
> 
> https://github.com/Manta-Network/Manta/pull/180/files#diff-84486036e95f7e22168734603cbdd311736c85fb23a0d0e0660988a67d3605ecR1315
> 
> I think we should make it not hardcoded (maybe with the exception of AWS instance types).

Made them env variables and now shared.

File: .github/workflows/publish-draft-releases.yml
Comment (ghzlatarev) [reply .github/workflows/publish-draft-releases.yml]: Done.

File: .github/resources/config-for-runtime-upgrade-test.json
Comment (stechu) [inline .github/resources/config-for-runtime-upgrade-test.json]: please add a line break here.

**Comment by stechu** [comment]
@ghzlatarev  thanks for the PR. I think it is good to go after fixing the conflicts.

**Review by stechu — APPROVED** [review]

**Comment by ghzlatarev** [comment]
Closing this one in favor of https://github.com/Manta-Network/Manta/pull/189
In order to use verified commit.

**Pull request closed without merging**
